worker-queue-rabbit/src/main/java/com/hpe/caf/worker/queue/rabbit/RabbitTaskInformation.java

```diff
@@ -0,0 +1,125 @@
+
+public class RabbitTaskInformation implements TaskInformation {
+    private final String inboundMessageId;
+    private final Object responseCountLock;
+    private volatile boolean negativeAckSent; 
+    private volatile int responseCount;
+    private volatile boolean isResponseCountFinal;
+    private final Object acknowledgementCountLock;
+    private volatile int acknowledgementCount;
+    private static final Logger LOG = LoggerFactory.getLogger(RabbitTaskInformation.class);
+
+    public RabbitTaskInformation(final String inboundMessageId) {
+        this.inboundMessageId = inboundMessageId;
+        this.responseCountLock = new Object();
+        this.responseCount = 0;
+        this.isResponseCountFinal = false;
+        this.acknowledgementCountLock = new Object();
+        this.acknowledgementCount = 0;
+        this.negativeAckSent=false;
+    }
+
+    @Override
+    public String getInboundMessageId() {
+        return inboundMessageId;
+    }
+
+    @Override
+    public void incrementResponseCount(final boolean isFinalResponse) {
+        synchronized (responseCountLock) {
+            if (isResponseCountFinal) {
+                throw new RuntimeException("Final response already set!");
+            }
+
+            responseCount++;
+
+            if (isFinalResponse) {
+                isResponseCountFinal = true;
+            }
+        }
+    }
+
+    /**
+     * Indicate that there are no more responses to come.
+     *
+     * @return true if all the responses have already been acknowledged
+     */
+    public boolean finalizeResponseCount()
+    {
+        synchronized (acknowledgementCountLock) {
+            synchronized (responseCountLock) {
+                isResponseCountFinal = true;
+            }
+
+            return responseCount == acknowledgementCount;
+        }
+    }
+
+    /**
+     * Increment the count of acknowledgements.
+     *
+     * @return true if this increment means that all responses have been acknowledged
+     */
+    public boolean incrementAcknowledgementCount()
+    {
+        synchronized (acknowledgementCountLock) {
+            final int ackCount = ++acknowledgementCount;
+
+            return isResponseCountFinal
+                    && (responseCount == ackCount);
+        }
+    }
+
+    public boolean areAllResponsesAcknowledged()
+    {
+        int response=getFinalResponseCount();
+        if(response<=0)
+        {
+            return false;
+        }
+        return getFinalResponseCount() <= acknowledgementCount;
+        
+    }
+
+    private int getFinalResponseCount()
+    {
+        synchronized (responseCountLock) {
+            if (!isResponseCountFinal) {
+                LOG.debug("Final response count is not known yet!");
+                return -1;
+            }
+
+            return responseCount;
+        }
+    }
+
+    public boolean isNegativeAckSent()
+    {
+        return negativeAckSent;
+    }
+
+    public void markNegativeAckAsSent()
```

Comment:
We talked about a better name for this. isTaskRejected setTaskRejected

Reply:
removed the setter and gave the name markNegativeAckAsSent(), thinking that to be a good enough name.

Reply:
To me it's confusing, markNegativeAckAsSent when in reality a negativeAck was received.